Documentation/installation/k8s-install-helm.rst

```diff
@@ -160,6 +160,11 @@ Install Cilium
                      && iptables -t nat -F AWS-CONNMARK-CHAIN-0 \\
                      && iptables -t nat -F AWS-CONNMARK-CHAIN-1
 
+         If ``enableIPv4Masquerade`` is set to ``true`` in the ENI environment 
+         (the default value is ``false`` starting from version 1.18, and ``true`` in earlier versions), 
+         the pod traffic to its own VPC will not be masqueraded, since Cilium treats
+         the entire VPC CIDR as the cluster range.
```

Comment:
Is there some specific integration with AWS to provide this or is this dependent on the user providing the native routing CIDR?

Reply:
this is because we generate the following rule, that will cause the traffic to anything in the VPC will skip the masquerading. It's a default behavior from us. nothing to do with AWS interation.  I put more explaniaon in the PR description.
```
-A CILIUM_POST_nat ! -d 192.168.0.0/16 -o eth+ -m comment --comment "cilium masquerade non-cluster" -j MASQUERADE
```

Reply:
Probably to confirm for sure we'd need to trace down where `state.localNodeInfo.ipv4AllocCIDR` is populated. Seems like it comes from the `CiliumNode` resource.

Reply:
Part of where I'm coming from here is that we aim to make Cilium as platform-agnostic as we can. If we can make a statement in the masquerading docs that describes generically how Cilium performs this masquerading, then that statement can benefit all users, not just AWS. On the other hand, if there is something specific to AWS then it can be useful to also highlight that. For instance we might describe that in general "Cilium uses the native routing CIDR if configured, or falls back to the node-specific Pod IPAM range", then we can add a platform-specific note that "on AWS the node-specific IPAM range is autodetected as the full cluster CIDR" (assuming we confirm that for sure).

Reply:
I have looked into the codebase and throw a debugger to confirm  codepath.

In cloud ENI mode(https://github.com/cilium/cilium/blob/f906eab6c0d63be93e7c48538b1c538010d339bb/pkg/ipam/ipam.go#L119-L120), if the native routing is not set. We will read Primary CIDR and assign that to the native routing

(https://github.com/cilium/cilium/blob/f906eab6c0d63be93e7c48538b1c538010d339bb/pkg/ipam/crd.go#L277-L308)

Reply:
Ah awesome, so we can confirm that this applies to a range of clouds :+1: good find.